### PR TITLE
Minor improvements to docs in std::env structures/functions.

### PR DESCRIPTION
**Pull request by frewsxcv**
* Call functions "functions" instead of "methods".
* Link structures to their constructor functions
* Add other misc. documentation links

### CODE REVIEW

File: src/libstd/env.rs [hunk 1]
```diff
@@ -68,15 +68,17 @@ pub fn set_current_dir<P: AsRef<Path>>(p: P) -> io::Result<()> {
 
 /// An iterator over a snapshot of the environment variables of this process.
 ///
-/// This iterator is created through `std::env::vars()` and yields `(String,
-/// String)` pairs.
+/// This structure is created through the [`std::env::vars`] function.
```
Comment (nagisa) [inline src/libstd/env.rs]: Previous string was strictly more informative IMO.

File: src/libstd/env.rs [hunk 2]
```diff
@@ -68,15 +68,17 @@ pub fn set_current_dir<P: AsRef<Path>>(p: P) -> io::Result<()> {
...
+///
+/// [`std::env::vars`]: fn.vars.html
 #[stable(feature = "env", since = "1.0.0")]
 pub struct Vars { inner: VarsOs }
 
 /// An iterator over a snapshot of the environment variables of this process.
 ///
-/// This iterator is created through `std::env::vars_os()` and yields
-/// `(OsString, OsString)` pairs.
+/// This structure is created through the [`std::env::vars_os`] function.
```
Comment (nagisa) [inline src/libstd/env.rs]: Ditto here.

File: src/libstd/env.rs (hunk 1 above)
Comment (frewsxcv) [inline src/libstd/env.rs]: How so? "yields `(String, String)` pairs." isn't saying anything more than what's displayed in the `Iterator` section of `Env`.

**Comment by brson** [comment]
r=me whenever @nagisa is satisfied

File: src/libstd/env.rs (hunk 1 above)
Comment (nagisa) [inline src/libstd/env.rs]: Oh, never mind me, I totally thought this was documentation for the function, which hasn’t such information displayed.

**Comment by nagisa** [comment]
LGTM.

**Comment by alexcrichton** [comment]
@bors: retry

* netbsd network failure #39097

**Pull request merged**